Sources/Diagnostics/DiagnosticsTracker.swift

```diff
@@ -36,30 +36,35 @@ protocol DiagnosticsTrackerType {
                                    responseCode: Int,
                                    backendErrorCode: Int?,
                                    resultOrigin: HTTPResponseOrigin?,
-                                   verificationResult: VerificationResult) async
+                                   verificationResult: VerificationResult)
 
 }
 
 @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
-final class DiagnosticsTracker: DiagnosticsTrackerType {
+final class DiagnosticsTracker: DiagnosticsTrackerType, Sendable {
 
     private let diagnosticsFileHandler: DiagnosticsFileHandlerType
+    private let diagnosticsDispatcher: OperationDispatcher
     private let dateProvider: DateProvider
 
     init(diagnosticsFileHandler: DiagnosticsFileHandlerType,
+         diagnosticsDispatcher: OperationDispatcher = .default,
          dateProvider: DateProvider = DateProvider()) {
         self.diagnosticsFileHandler = diagnosticsFileHandler
+        self.diagnosticsDispatcher = diagnosticsDispatcher
         self.dateProvider = dateProvider
     }
 
-    func track(_ event: DiagnosticsEvent) async {
-        await self.clearDiagnosticsFileIfTooBig()
-        await self.diagnosticsFileHandler.appendEvent(diagnosticsEvent: event)
+    func track(_ event: DiagnosticsEvent) {
+        self.diagnosticsDispatcher.dispatchOnWorkerThread {
```

Comment:
This is the main change, now dispatching to a background thread happens internally.

Reply:
I don't think you need a `diagnosticsDispatcher`? You could just create a Task here,

```swift
func track(_ event: DiagnosticsEvent) {
    Task {
        await self.clearDiagnosticsFileIfTooBig()
        await self.diagnosticsFileHandler.appendEvent(diagnosticsEvent: event)
    }
}
```

However, with this change, events submitted to `track()` aren't guaranteed to run sequentially as they are with the `OperationDispatcher`. But this guarantee won't have existed before, so unless you want to make that change, I think using `Task` is the way to go.

Reply:
There are 2 main reasons I decided to go with the OperationDispatcher:
- First, as you said, this will make the operations run sequentially, which I believe would be safer, since there could be chances of changing contexts between tracking different events, and if we reach the file too big issue, we might be missing extra data. As you said, this problem would be present right now.
- Second, I wanted to be able to test this class. If I used `Task`, I don't have a good way to determine when the tracking finished to do the assertions. With OperationDispatcher, we can pass a MockOperationDispatcher in the tests that runs things synchronously (mostly), allowing us to do the assertions immediately.

Lmk if you would like to talk more about it!

Reply:
OK, sounds good!